### PR TITLE
[Docs/Move] Reword

### PR DESCRIPTION
**Pull request by amnn**
## Description

This one's a large rewrite:

- Refer to "Core Move" and "Diem Move" as "Move on Diem", everywhere.
- Reword section explaining scaling issues with global storages and contracts that store large mappings, and the explanation of how global storage works -- the original explanations didn't really seem to hold whatever for me.
- Added a link to the `entry` function docs.

## Test Plan

:eyes:

### CODE REVIEW

File: docs/content/concepts/sui-move-concepts/move-on-sui.mdx [hunk 1]
```diff
@@ -28,33 +28,30 @@ Key differences with Move on Sui include:
 
 ### Object-centric global storage {#global-storage}
 
-In Diem-style Move, global storage is part of the programming model that you can access through special operations, such as `move_to`, `move_from`, and more global storage operators. Core Move stores both resources and modules in global storage. When you publish a module, it's stored into a newly generated module address inside Move. When you create a new object (resource), it's usually stored into some address, as well. This approach creates a problem, however, as on-chain storage is expensive and limited (not optimized for storage and indexing). Current blockchains cannot scale effectively to handle storage-heavy applications, such as marketplaces and social apps.
```
Comment (amnn) [inline docs/content/concepts/sui-move-concepts/move-on-sui.mdx:31]: @sblackshear, @tnowacki, this passage didn't really match my understanding of how addressing/resources/modules worked on Diem, and what the scaling pitfalls were with that, but I wasn't around at the time -- could you check my working here?

**Review by ronny-mysten — APPROVED** [review]

File: docs/content/concepts/sui-move-concepts/move-on-sui.mdx [hunk 2]
```diff
@@ -28,33 +28,30 @@ Key differences with Move on Sui include:
...
-
-Move on Sui addresses the scaling issue by not having global storage. Consequently, Move doesn't allow any of the global storage-related operations (there is a bytecode verifier to detect violations for this). Instead, storage happens exclusively within Sui. When you publish a module, the newly published module is stored in Sui storage, instead of Move storage. Similarly, newly created objects are stored in Sui storage.
-
-:::info
+In Move on Diem, global storage is part of the programming model. Resources and modules are held in global storage, owned by an account which has an address. Transactions that access resources are only required to specify the types of resources being accessed, and not the accounts they are being accessed from, which can be specified dynamically, using special operations such as `move_to` and `move_from`.
```
Comment (tnowacki) [inline docs/content/concepts/sui-move-concepts/move-on-sui.mdx:31]: > Transactions that access resources are only required to specify the types of resources being accessed, and not the accounts they are being accessed from

This was sort of true at a time in Diem, but also not really.

Move on Diem had no fundamental requirements for any types of accesses. There were some safety considerations in modules, but from a transaction standpoint, anything could be accessed.

File: docs/content/concepts/sui-move-concepts/move-on-sui.mdx [hunk 3]
```diff
@@ -28,33 +28,30 @@ Key differences with Move on Sui include:
...
 
 ### Object with key ability, globally unique IDs {#global-unique}
 
-In Diem-style Move, the `key` ability is used to tell that the type can be used as a key for global storage. In Sui, this ability has been repurposed to declare a struct as an object. Objects owned by an account are passed in as arguments to transactions, and other objects (using the dynamic fields API dynamic_fields.move) can be accessed dynamically during the transaction. Sui requires that any struct with key ability must start with an `id` field with the UID type. The UID type contains both the `ObjectID`. Other metadata (such as the sequence number) is managed by the Sui network outside of Move. Sui has a bytecode verifier in place to make sure that the UID field cannot be transferred to other objects (the IDs of objects cannot be re-used once the object is destructured).
+In Move on Diem, the `key` ability indicates that the type (along with an account address) can be used as a key in global storage.
```
Comment (tnowacki) [inline docs/content/concepts/sui-move-concepts/move-on-sui.mdx:45]: Maybe add something about resources
```suggestion
In Move on Diem, the `key` ability indicates that the type (along with an account address) can be used as a key in global storage, i.e. the type is a resource. 
```
or something similar

File: docs/content/concepts/sui-move-concepts/move-on-sui.mdx [hunk 4]
```diff
@@ -28,33 +28,30 @@ Key differences with Move on Sui include:
...
+
+On Sui, this ability indicates that a struct is an object type and comes with an additional requirement that the first field of the struct has signature `id: UID`, to contain the object's unique address on-chain. Sui's bytecode verifier ensures that new objects are always assigned fresh `UID`s (identifiers are never re-used).
```
Comment (tnowacki) [inline docs/content/concepts/sui-move-concepts/move-on-sui.mdx:47]:
```suggestion
On Sui, this `key` ability indicates that a struct is an object type and comes with an additional requirement that the first field of the struct has signature `id: UID`, to contain the object's unique address on-chain. Sui's bytecode verifier ensures that new objects are always assigned fresh `UID`s (identifiers are never re-used).
```

File: docs/content/concepts/sui-move-concepts/move-on-sui.mdx [hunk 5]
```diff
@@ -28,33 +28,30 @@ Key differences with Move on Sui include:
...
-
-In addition to calling public functions, you can call a function that is marked `entry` even if it is private. As long as other public functions have not used its inputs. 
+You can call public functions from Sui transactions (called programmable transaction blocks). These functions can take objects by value, by immutable reference, or by mutable reference. If taken by value, you can destroy the object, wrap it (in another object), or transfer it (to a Sui ID specified by an address). If taken by mutable reference, the modified version of the object saves to storage without any change in ownership. In any case, the Sui network authenticates the object and declares it's usage as a part of the transaction.
 
+In addition to calling public functions, you can call a function that is marked [entry](entry-functions.mdx) even if it is private as long as other public functions have not used its inputs.
```
Comment (tnowacki) [inline docs/content/concepts/sui-move-concepts/move-on-sui.mdx:57]:
```suggestion
In addition to calling public functions, you can call a function that is marked [entry](entry-functions.mdx) even if it is private as long as other non-`entry` functions have not used its inputs.
```

File: docs/content/concepts/sui-move-concepts/move-on-sui.mdx (hunk 2 above)
Comment (amnn) [reply docs/content/concepts/sui-move-concepts/move-on-sui.mdx:31]: Ah, so the `acquires` annotations on functions were not used for transaction scheduling?

**Pull request merged**